sdlib/d/gc/region.d

```diff
@@ -63,6 +66,9 @@ private:
 	size_t minAddress = AddressSpace;
 	size_t maxAddress = 0;
 
+	// Count of blocks that are currently acquired.
+	ValueTracker!long nBlocks =
+		ValueTracker!long(0, MinimumBlockCollectThreshold);
```

Comment:
I don't really understand the problem the value tracker is supposed to solve.

To me this is a step backward from what we had before. The RegionAllocator is still in charge of triggering GC collection but it's been obfuscated through a layer of abstraction.

While tracking the number of blocks make sense for the RegionAllocator, looking at the structure that is emerging here clearly indicates that the same isn't true for the GC thresold. It is touched in 2 methods, `checkBlockThreshold` and `setBlockThreshold`. Both methods do not touch anything else in the RegionAllocator but the block count.

Whatever is setting the thresold in the region allocator, should keep it for itself. It's absurd that this whatever has to put the info in the region allocator, then check its value through the region allocator, when nothing in the region allocator require this threshold to begin with.